### PR TITLE
Feature/future header

### PR DESCRIPTION
**Pull request by lbebber**
#### What is the purpose of this pull request?
First, unstable implementation of the new composable header

Also adds stubs for some components which will be further developed in the future.

Also adds support for the `unstable--menu` block on the now legacy header.

#### Types of changes
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/legacy/index.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
+import React, { Fragment, useEffect, useLayoutEffect, useState } from 'react'
```
Comment (lbebber) [inline react/legacy/index.js]: This file was just moved from `react` to `react/legacy`, please refrain from looking too deeply into it for this PR

File: react/legacy/index.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
+import React, { Fragment, useEffect, useState } from 'react'
```
Comment (lbebber) [inline react/legacy/index.js]: This file was mostly just moved from `react` to `react/legacy`, please refrain from delving too deeply into it for this PR

File: react/components/CustomHeader.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        {matches => matches ? (
+          <ExtensionPoint id="unstable--header-layout.mobile" />
+        ) : (
+          <ExtensionPoint id="unstable--header-layout.desktop" />
+        )}
+      </Media>
+    </React.Fragment>
+  )
+}
+
```
Comment (rerissondaniel) [inline react/components/CustomHeader.js]:
```suggestion

CustomHeader.schema = {
  title: 'editor.header.title',
  description: 'editor.header.description',
}
```

File: react/components/Row.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+const Row = ({ children, fullWidth, sticky, inverted }) => {
+  const content = (
+    <div className="w-100 flex items-center">
+      {children}
+    </div>
+  )
+
+  return (
+    <div
+      className={`w-100 top-0 ${sticky ? 'z-999' : '' } ${inverted ? 'bg-base--inverted c-on-base--inverted' : 'bg-base c-on-base'}`}
```
Comment (matheusps) [inline react/components/Row.js]: You could use `classNames` here

File: react/index.js [hunk 5]
```diff
@@ -1,140 +1,19 @@
...
-      title: 'editor.header.show.login.title',
-      type: 'boolean',
-      default: true,
-      isLayout: true,
-    },
-  },
-}
-
-export default Header
+export default Header
```
Comment (rerissondaniel) [inline react/index.js]:
```suggestion
Header.schema = {
  title: 'editor.header.title',
  description: 'editor.header.description',
}

export default Header
```

**Pull request merged**